src/core/electrostatics_magnetostatics/p3m.cpp

```diff
@@ -95,22 +95,13 @@ p3m_data_struct p3m;
  */
 static void p3m_init_a_ai_cao_cut();
 
-/** Realloc charge assignment fields. */
-static void p3m_realloc_ca_fields(int newsize);
-
 static bool p3m_sanity_checks_system(const Utils::Vector3i &grid);
 
 /** Checks for correctness for charges in P3M of the cao_cut,
  *  necessary when the box length changes
  */
 static bool p3m_sanity_checks_boxl();
 
-/** Interpolate the P-th order charge assignment function from
- *  @cite hockney88a 5-189 (or 8-61). The following charge fractions
- *  are also tabulated in @cite deserno98a @cite deserno98b.
```

Comment:
I think these references are actually useful and should be moved to the docstring of
https://github.com/espressomd/espresso/blob/7f90057b87cb5b676a29fb42dbb7897157c04b95/src/utils/include/utils/math/bspline.hpp#L29-L30
This other function should be documented as well, looks like it's the first derivative:
https://github.com/espressomd/espresso/blob/7f90057b87cb5b676a29fb42dbb7897157c04b95/src/utils/include/utils/math/bspline.hpp#L189

Reply:
If we want to keep them they should go on `p3m_interpolate`, as they are p3m-specific citations, while bsplines are not, and they are used in different contexts...

Reply:
If the coefficients in `bspline()` haven't been derived specifically for charge interpolation, then you're right, we don't need to cite the P3M paper (deserno98a) from which they come from. I'm asking this because I couldn't find the same tabulated values in other documents describing the B-splines recursion formula.

Reply:
They are the centered ones (base interval `[-0.5, 0.5]` ), the formula from wikipedia is for base inveral `[0, 1] `. I think those base functions should be cleaned up anyway, but I think that is for another PR.